scipy/interpolate/_bsplines.py

```diff
@@ -0,0 +1,793 @@
+        >>> k
+        3
+
+        Construct a second order b-spline on ``[0, 1, 1, 2]``, and compare
+        to its explicit form:
+
+        >>> t = [-1, 0, 1, 1, 2]
+        >>> b = BSpline.basis_element(t[1:])
+        >>> def f(x):
+        ...     return np.where(x < 1, x*x, (2. - x)**2)
+
+        >>> import matplotlib.pyplot as plt
+        >>> fig, ax = plt.subplots()
+        >>> x = np.linspace(0, 2, 51)
+        >>> ax.plot(x, b(x), 'g', lw=3)
+        >>> ax.plot(x, f(x), 'r', lw=8, alpha=0.4)
+        >>> ax.grid(True)
+        >>> plt.show()
+
+        """
+        k = len(t) - 2
+        t = np.r_[(t[0]-1,) * k, t, (t[-1]+1,) * k]
+        c = np.zeros_like(t)
+        c[k] = 1.
+        return cls(t, c, k, extrapolate)
+
+    def __call__(self, x, nu=0, extrapolate=None):
+        """
+        Evaluate a spline function.
+
+        Parameters
+        ----------
+        x : array_like
+            points to evaluate the spline at.
+        nu: int, optional
+            derivative to evaluate (default is 0).
+        extrapolate : bool, optional
+            whether to extrapolate based on the first and last intervals
+            or return nans. Default is `self.extrapolate`.
+
+        Returns
+        -------
+        y : array_like
+            Shape is determined by replacing the interpolation axis
+            in the coefficient array with the shape of `x`.
+
+        """
+        if extrapolate is None:
+            extrapolate = self.extrapolate
+        x = np.asarray(x)
+        x_shape = x.shape
+        x = np.ascontiguousarray(x.ravel(), dtype=np.float_)
+        out = np.empty((len(x), prod(self.c.shape[1:])), dtype=self.c.dtype)
+        self._ensure_c_contiguous()
+        self._evaluate(x, nu, extrapolate, out)
+        return out.reshape(x_shape + self.c.shape[1:])
+
+    def _evaluate(self, xp, nu, extrapolate, out):
+        _bspl.evaluate_spline(self.t, self.c.reshape(self.c.shape[0], -1),
+                self.k, xp, nu, extrapolate, out)
+
+    def _ensure_c_contiguous(self):
+        """
+        c and t may be modified by the user. The Cython code expects
+        that they are C contiguous.
+
+        """
+        if not self.t.flags.c_contiguous:
+            self.x = self.x.copy()
+        if not self.c.flags.c_contiguous:
+            self.c = self.c.copy()
+
+    def derivative(self, nu=1):
+        """Return a b-spline representing the derivative.
+
+        Parameters
+        ----------
+        nu : int, optional
+            Derivative order.
+            Default is 1.
+
+        Returns
+        -------
+        b : BSpline object
+            A new instance representing the derivative.
+
+        See Also
+        --------
+        splder, splantider
+
+        """
+        c = self.c
+        # pad the c array if needed
+        ct = len(self.t) - len(c)
+        if ct > 0:
+            c = np.r_[c, np.zeros((ct,) + c.shape[1:])]
+        tck = fitpack.splder((self.t, c, self.k), nu)
```

Comment:
Indeed, done in https://github.com/ev-br/scipy/commit/446bf12e5458906cfb2bb27abc45708dd6f7e1b6 with some small cost in readability but hey.